Changed Wikihouse Plugin to be compatible with Sketchup Extensions.

MrKriss opened the pull request:
Main additions:
- `make` or `make build` now generates `wikihouse_extension.rbz` 
- `make release` _should_ upload following to Amazon S3 bucket:
  - `wikihouse_extension.rbz` and `wikihouse_extension-{git-rev}.rbz`
- Included `.gitignore` for .rbz files
- Renamed all instances of plugin to extension for clarity. 
- Updated install instruction in README

MrKriss committed:
First Commit with minor Comments added

MrKriss committed:
Restructured folders to hold all plugin code in a single folder, and a
short script to add the plugin as a SketchUp Extension. Also updated
README install instructions.

MrKriss committed:
Added .rbz file for extension install

MrKriss committed:
Minor changes to README

MrKriss committed:
Changed folder name to wikihouse-extension

MrKriss committed:
make or make build now generates wikihouse_extension.rbz

make release should upload following to Amazon S3
bucket: wikihouse_extension.rbz and wikihouse_extension-{git-rev}.rbz

Included .gitignore for .rbz files.

Renamed all instances of plugin to extension for clarity.

Updated install instruction in README

MrKriss committed:
Introduced an encapsulating Module 'WikihouseExtension' to allow for
namespace recalling between different files of ruby code.

Beginning to split up wikihouse.rb into smaller files of related code. 

Run flags and config constants now in wikihouse_extension_loader before
extension is loaded.

Extra constants now in constants.rb
Utility functions and classes now in utils.rb
Output writters now in writers.rb

SVG output currently has an error. Investigating.

MrKriss committed:
Fixed the error. Origin was extra white space in generated svg file.

MrKriss committed:
Minor comment edits + experimenting with SVG writer code.

MrKriss commented:
Hi Tav, whats the status of this? And should I continue to submit more work to this same pull request or wait till you are done with this one? I'm currently working on my own Dev branch of the folk.

thruflo commented:
/me pokes @tav

tav commented:
Hey @MrKriss, haven't forgotten you... just been busy with my upcoming wedding in 2 weeks! :smiley: 

Will review and merge it soonish. As for other work you are doing, it would be great if you could break it down into individual pull requests per feature. Thanks!

MrKriss committed:
Updated Make file to also install files into default location on Mac
with "install_mac"

MrKriss committed:
All Web dialogue will soon be moved here fore easier reference.

MrKriss committed:
New JSON module for parsing of Ruby Hashes and Arrays to and from json
object strings. Used in parsing data to web dialogues.

MrKriss committed:
File no longer needed. Constants moved into utils.rb or
wikihouse_extension_loader.rb

MrKriss committed:
Added some more constants

MrKriss committed:
Changed dimensions to a module vaiable. Added a Hash
'@@wikihouse_settings' to store variable info.

MrKriss committed:
Added settings Web Dialogue. Testing in progress.

MrKriss committed:
Finished Testing Settings Web Dialogue.

MrKriss committed:
Moved all Web Dialogue related code to one file.

MrKriss committed:
Make wikihouse_settings a global variable and added sheet_depth
parameter.

MrKriss committed:
Added Changes log file

MrKriss committed:
Updated + added alternative manual install

MrKriss committed:
Minor formating edits

MrKriss committed:
Minor markdown edits

MrKriss committed:
minor edits to markdown

MrKriss committed:
Fixed error in Web Dialogue save and error callbacks.

MrKriss committed:
Added install_mac_SU8 and install_mac_SU2013 options for make file.
These
will update all files in necessary plugin folders. Note folders must
already exist.

MrKriss committed:
Minor edits

MrKriss committed:
Changed JSON to a class instead of module. Seems there is a validation
check in SU 2013 that was throwing an error on loading plugin.
